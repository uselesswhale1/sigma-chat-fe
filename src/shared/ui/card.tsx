import { Box, Flex, Avatar, Heading } from "@chakra-ui/react";
import { PALLETE } from "../constants/color-pallete";

interface CardProps {
  bg?: string;
  name: string;
  photoUrl: string;
  header?: string;
  text: string;
  onClick?: () => void;
  actions?: JSX.Element;
  isReversed?: boolean;
}

export const Card = ({
  bg = PALLETE.bg,
  name,
  photoUrl,
  header,
  text,
  actions,
  isReversed,
  onClick,
}: CardProps): JSX.Element => {
  return (
    <Flex
      p={2}
      m={1}
      bgColor={bg}
      gap="4"
      alignItems="center"
      flexWrap="wrap"
      onClick={onClick}
      flexDirection={isReversed ? "row-reverse" : "row"}
    >
      <Box>
        <Avatar name={name} src={photoUrl} />
      </Box>

      <Box>
        <Heading size="sm" textAlign="left" children={header} />
        <Box
          w={200}
          whiteSpace="nowrap"
          overflow="hidden"
          textOverflow="ellipsis"
          children={text}
        />
      </Box>

      <Box>{actions}</Box>
    </Flex>
  );
};
