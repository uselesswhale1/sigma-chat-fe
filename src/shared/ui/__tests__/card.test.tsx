import { render } from "@testing-library/react";
import { Card } from "../card";

describe("Card should", () => {
  const props = {
    name: "name",
    text: "message",
    header: "header",
  };

  it("match snapshot", () => {
    const { container } = render(<Card {...props} />);

    expect(container).toMatchSnapshot();
  });

  it("match snapshot when reversed", () => {
    const { container } = render(<Card {...props} isReversed />);

    expect(container).toMatchSnapshot();
  });
});
