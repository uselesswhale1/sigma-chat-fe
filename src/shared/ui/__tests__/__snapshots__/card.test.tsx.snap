// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
<div>
  <div
    class="css-1utkaw9"
  >
    <div
      class="css-0"
    >
      <span
        class="chakra-avatar css-1ruhx4w"
      >
        <div
          aria-label="name"
          class="chakra-avatar__initials css-0"
          role="img"
        >
          n
        </div>
      </span>
    </div>
    <div
      class="css-0"
    >
      <h2
        class="chakra-heading css-13brihr"
      >
        header
      </h2>
      <div
        class="css-1a1loib"
      >
        message
      </div>
    </div>
    <div
      class="css-0"
    />
  </div>
</div>
`;

exports[`Card should match snapshot when reversed 1`] = `
<div>
  <div
    class="css-1a6f4eo"
  >
    <div
      class="css-0"
    >
      <span
        class="chakra-avatar css-1ruhx4w"
      >
        <div
          aria-label="name"
          class="chakra-avatar__initials css-0"
          role="img"
        >
          n
        </div>
      </span>
    </div>
    <div
      class="css-0"
    >
      <h2
        class="chakra-heading css-13brihr"
      >
        header
      </h2>
      <div
        class="css-1a1loib"
      >
        message
      </div>
    </div>
    <div
      class="css-0"
    />
  </div>
</div>
`;
